sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -130,11 +131,39 @@ export class GlobalEndpointManager {
     return canUse;
   }
 
+  public getEffectiveExcludedLocations(
+    options: RequestOptions | FeedOptions | ChangeFeedIteratorOptions = {},
+    resourceType: ResourceType,
+  ): Set<string> {
+    if (!canApplyExcludedLocations(resourceType)) {
+      return new Set();
+    }
+    const excludedLocations = options?.excludedLocations;
+
+    return excludedLocations?.length
+      ? new Set(excludedLocations.map(normalizeEndpoint))
+      : new Set();
+  }
+
+  public filterExcludedLocations(
+    preferredLocations: string[],
+    excludedLocations?: Set<string>,
+  ): string[] {
+    if (!excludedLocations || excludedLocations.size === 0) {
+      return preferredLocations;
+    }
+    const filteredLocations = preferredLocations.filter((location) => {
+      return !excludedLocations.has(normalizeEndpoint(location));
+    });
+    return filteredLocations;
+  }
+
   public async resolveServiceEndpoint(
     diagnosticNode: DiagnosticNodeInternal,
     resourceType: ResourceType,
     operationType: OperationType,
     startServiceEndpointIndex: number = 0, // Represents the starting index for selecting servers.
+    requestOptions: RequestOptions | FeedOptions | ChangeFeedIteratorOptions = {}, // add to support request-level excluded region(location)
```

Comment:
Why do we have default value for options here. Do we have any caller where we are not passing this?